file.go

```diff
@@ -134,10 +137,13 @@ func (f *win32File) Close() error {
 // prepareIo prepares for a new IO operation.
 // The caller must call f.wg.Done() when the IO is finished, prior to Close() returning.
 func (f *win32File) prepareIo() (*ioOperation, error) {
+	f.wgLock.RLock()
 	if f.closing.isSet() {
```

Comment:
Note I chose to keep the atomic bool here as it is accessed in multiple locations that are not affected by this race and think that adding the lock around each access would likely be higher perf impact than an atomic bool and lock in this one case. I didn't test this theory though, so I might be wrong.